dbms/src/Common/MPMCQueue.h

```diff
@@ -73,9 +73,21 @@ class MPMCQueue
 public:
     using Status = MPMCQueueStatus;
     using Result = MPMCQueueResult;
+    using ElementAuxiliaryMemoryUsageFunc = std::function<size_t(const T & element)>;
 
     explicit MPMCQueue(size_t capacity_)
         : capacity(capacity_)
+        , max_auxiliary_memory_usage(std::numeric_limits<size_t>::max())
+        , get_auxiliary_memory_usage([](const T &) { return 0; })
```

Comment:
How about using templates?
```
template <typename T>
struct DefaultGetAuxiliaryMemoryUsage
{
    size_t get(const T &) { return 0; }
};
template <typename T, typename GetAuxiliaryMemoryUsage = DefaultGetAuxiliaryMemoryUsage>
class MPMCQueue
```

Reply:
Seems don't have too much benifits to use template here?

Reply:
ok